docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,193 @@
+-------------------------------
 
-If your dataset is relatively small or the underlying estimator doesn't implement
-``partial_fit``, you can use :class:`dask_ml.model_selection.GridSearchCV` or
-:class:`dask_ml.model_selection.RandomizedSearchCV`.
-These are drop-in replacements for their scikit-learn counterparts, that should offer better performance and handling of Dask Arrays and DataFrames.
-The underlying estimator will need to be able to train on each cross-validation split of the data.
-See :ref:`hyperparameter.drop-in` for more.
+Dask-ML has a ton of tools to get around common issues in hyperparameter
+optimization:
 
-If your data is large and the underlying estimator implements ``partial_fit``, you can
-Dask-ML's :ref:`*incremental* hyperparameter optimizers <hyperparameter.incremental>`.
+.. autosummary::
+   dask_ml.model_selection.GridSearchCV
+   dask_ml.model_selection.RandomizedSearchCV
+   dask_ml.model_selection.IncrementalSearchCV
+   dask_ml.model_selection.HyperbandSearchCV
+   dask_ml.model_selection.SuccessiveHalvingSearchCV
+
+These avoid the two most common issues in hyperparameter optimization, when
+the hyperparameter search is...
+
+1. "**memory constrained"**, which happens when the dataset size is too large
+   to fit in memory.  This typically happens when a model needs to be tuned for
+   a larger-than-memory dataset after local development.
+2. "**compute constrained**", which happen when the computation takes too long
+   even with data that can fit in memory.  This typically happens when many
+   hyperparameters need to be tuned, which especially common with neural
+   networks.
+
+Here's an example of what "memory constrained" means:
+
+.. code:: python
+
+   # not memory constrained
+   import pandas as pd
+   df = pd.read_csv("0.parquet")
+   print(df.shape)  # (30000, 200) => 23MB
+
+   # memory constrained
+   import dask.dataframe as dd
+   # Read 1000 of the above dataframes (22GB of data)
+   ddf = dd.read_parquet("*.parquet")
+
+Here's an example of what "compute constrained" means. Let's look at when
+
+.. code:: python
+
+   from scipy.stats import uniform, loguniform
+   from sklearn.linear_model import SGDClasifier
+
+   model = SGDClasifier()
+
+   # not compute constrained
+   params = {"l1_ratio": uniform(0, 1)}
+
+   # compute constrained
+   params = {
+       "l1_ratio": uniform(0, 1),
+       "alpha": loguniform(1e-5, 1e-1),
+       "penalty": ["l2", "l1", "elasticnet"],
+       "learning_rate": ["invscaling", "adaptive"],
+       "power_t": uniform(0, 1),
+       "average": [True, False],
+   }
+
+These issues are independent and both can happen the same time. Dask-ML has
+tools to address all 4 combinations. Let's look at each case.
+
+Neither compute nor memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when there aren't many hyperparameters to tune and the data
+fits in memory. This is common when the search can run on your laptop.
```

Comment:
Maybe strike the "on your laptop" and replace it with something like "doesn't take too long to train." Many groups use single large nodes, which works well if you have that option.